### PR TITLE
[DPE-3231] Kafka slim image using rockcraft

### PR DESCRIPTION
**Pull request by deusebio**
Providing Rock implementation for slim image. I have also spent some time to create some tests for the image that we could also use in the charm image (that is currently not tested :( )

### CODE REVIEW

File: rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  [Apache Zookeeper image](https://hub.docker.com/r/ubuntu/zookeeper)
+  to run properly.
+license: Apache-2.0  # your application's SPDX license
+
+platforms:  # The platforms this ROCK should be built on and run on
+  amd64:
+
+environment:
+  JAVA_HOME: /usr/lib/jvm/java-18-openjdk-amd64
+  ZOOKEEPER_HOST: "zookeeper"
```
Comment (zmraul) [inline rockcraft.yaml:27]: **question:** is this just a placeholder?

**Review by zmraul — APPROVED** [review]
LGTM. Why are metrics exporters missing from the rock? Are they not considered important part of a slim image?

File: rockcraft.yaml (hunk 1 above)
Comment (deusebio) [reply rockcraft.yaml:27]: Not really. I mean, the pebble Kafka service needs these (see command in line 37). If I were to put that on the environment of Pebble, that would not make it possible to override it. Overriding this works now just fine, which I also test [here](https://github.com/canonical/kafka-rock/pull/2/files#diff-2465a5fd98a33922a6194b3483e771e3ab22469b6fcd9ba4ead7007923361475R64).

The default is chosen such that in a simple docker deployment (where in the docker network is also deployed a zookeeper named `zookeeper` which is the name of the ubuntu zookeeper slim image) this works out of the box (see [here](https://github.com/canonical/kafka-rock/pull/2/files#diff-2465a5fd98a33922a6194b3483e771e3ab22469b6fcd9ba4ead7007923361475R50)).

**Comment by deusebio** [comment]
> Why are metrics exporters missing from the rock? Are they not considered important part of a slim image?

Yes! Slim image should really just be the bare minimum. The idea is that other people can take the image and extends it for their more specific use-cases. Same approach was taken for ZK

**Review by marcoppenheimer — APPROVED** [review]

**Pull request merged**